homeassistant/components/vacuum/roomba.py

```diff
@@ -0,0 +1,271 @@
+"""
+Support for Wi-Fi enabled iRobot Roombas.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.roomba/
+"""
+import asyncio
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.vacuum import (
+    VacuumDevice,
+    PLATFORM_SCHEMA, SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_PAUSE,
+    SUPPORT_STOP, SUPPORT_RETURN_HOME, SUPPORT_BATTERY, SUPPORT_STATUS,
+    SUPPORT_SEND_COMMAND)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, CONF_NAME, CONF_HOST, CONF_USERNAME, CONF_PASSWORD)
+
+REQUIREMENTS = ['roombapy==1.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BIN_FULL = 'bin_full'
+ATTR_BIN_PRESENT = 'bin_present'
+ATTR_CLEANING_TIME = 'cleaning_time'
+ATTR_CLEANED_AREA = 'cleaned_area'
+ATTR_ERROR = 'error'
+ATTR_POSITION = 'position'
+ATTR_SOFTWARE_VERSION = 'software_version'
+
+CONF_CERT = 'certificate'
+CONF_CONTINUOUS = 'continuous'
+
+DEFAULT_CERT = '/etc/ssl/certs/ca-certificates.crt'
+DEFAULT_CONTINUOUS = True
+DEFAULT_NAME = 'Roomba'
+
+ICON = 'mdi:roomba'
+PLATFORM = 'roomba'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_CERT, default=DEFAULT_CERT): cv.string,
+    vol.Optional(CONF_CONTINUOUS, default=DEFAULT_CONTINUOUS): cv.boolean,
+}, extra=vol.ALLOW_EXTRA)
+
+SUPPORT_ROOMBA = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PAUSE | \
+                 SUPPORT_STOP | SUPPORT_RETURN_HOME | SUPPORT_BATTERY | \
+                 SUPPORT_STATUS | SUPPORT_SEND_COMMAND
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the iRobot Roomba vacuum cleaner platform."""
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    certificate = config.get(CONF_CERT)
+    continuous = config.get(CONF_CONTINUOUS)
+
+    # Create handler
+    roomba = RoombaVacuum(
```

Comment:
But I'd still need to pass the `hass` object so that I can issue `yield from self.hass.async_add_job()` from within the `RoombaVacuum` class